electron-bridge: Add decrypt_clipboard helper (for browser-based social login)

andersk opened the pull request:
This one helper allows us to implement browser-based social login entirely on the server side. See zulip/zulip#14747.

andersk committed:
electron-bridge: Add decrypt_clipboard helper.

This one helper allows us to implement browser-based social login
entirely on the server side.

Signed-off-by: Anders Kaseorg <anders@zulip.com>

timabbott commented on app/renderer/js/electron-bridge.ts:
```diff
@@ -46,6 +47,28 @@ class ElectronBridge extends EventEmitter {
+						{ authTagLength: 16 }
+					);
+					decipher.setAuthTag(ciphertext.slice(-16));
+					return decipher.update(ciphertext.slice(12, -16), undefined, 'utf8') + decipher.final('utf8');
```
I would probably extract variables for the `ciphertext.slice()` sections all the start for readability, e.g. `const iv = ciphertext.slice(0, 12);`, etc.

andersk replied:
Done.

timabbott commented on app/renderer/js/electron-bridge.ts:
```diff
@@ -46,6 +47,28 @@ class ElectronBridge extends EventEmitter {
 	};
+
+	decrypt_clipboard = (): { key: string; check(): string | null } => {
+		const key = crypto.randomBytes(32);
```
I think probably it's worth a block comment here explaining the interface of this, namely that it provides a randomly generated key as well as a function to decrypt content encrypted using that key; and the encryption algorithm used by the server needs to be the same as what is used here.  

I sorta wonder if we should have this present a third parameter, `algorithm_version`, that is an int that gets passed on to the server; this would be setup for if we decide to change the algorithm in the future, the server could know how the app is expecting us to encrypt.  I suppose we could always add this later with "didn't send" meaning v0, but it'd be pretty cheap to just have an `algorithm_version: 1` here so that the server-side code can assume that field is always present.

(Even if it's ignored for now, the server-side JS could always change to use it)

andersk replied:
Done.

andersk commented:
Also moved this helper to a new file, and changed the polling interface to a promise interface in case we want to do something more clever than `setInterval(…, 1000)` in the future.

timabbott commented:
This lgtm.

andersk commented:
Updated to poll the clipboard only when the app has focus.

timabbott commented:
The polling logic lgtm; feel free to merge.
